evalml/tests/component_tests/test_imputer.py

```diff
@@ -327,3 +335,31 @@ def test_imputer_multitype_with_one_bool(data_type, make_data_type):
     imputer.fit(X_multi, y)
     X_multi_t = imputer.transform(X_multi)
     assert_frame_equal(X_multi_expected_arr, X_multi_t.to_dataframe())
+
+
+@pytest.mark.parametrize("X_df", [pd.DataFrame(pd.Series([1, 2, 3], dtype="Int64")),
+                                  pd.DataFrame(pd.Series([1., 2., 3.], dtype="float")),
+                                  pd.DataFrame(pd.Series(['a', 'b', 'a'], dtype="category")),
+                                  pd.DataFrame(pd.Series([True, False, True], dtype="boolean")),
+                                  pd.DataFrame(pd.Series(['this will be a natural language column because length', 'yay', 'hay'], dtype="string"))])
+@pytest.mark.parametrize("has_nan", [True, False])
+def test_imputer_woodwork_custom_overrides_returned_by_components(X_df, has_nan):
+    y = pd.Series([1, 2, 1])
+    if has_nan:
+        X_df.iloc[len(X_df) - 1, 0] = np.nan
+    override_types = [Integer, Double, Categorical, NaturalLanguage, Boolean]
+    for logical_type in override_types:
+        try:
+            X = ww.DataTable(X_df, logical_types={0: logical_type})
+        except TypeError:
+            continue
+
+        imputer = Imputer()
+        imputer.fit(X, y)
+        transformed = imputer.transform(X, y)
+        assert isinstance(transformed, ww.DataTable)
+
+        if logical_type == Integer:
+            assert transformed.logical_types == {0: Double}
```

Comment:
How come integers are never carried over? I understand that if  if the numeric_impute_strategy is "mean" then you can get doubles at the end. But what if the impute strategy is mode?

Reply:
Hehe, you're right, this is dependent on impute strategy. Going to update `test_simple_imputer_woodwork_custom_overrides_returned_by_components` and `test_imputer_woodwork_custom_overrides_returned_by_components` to handle testing each of the different strategies. 😁